corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_view.js

```diff
@@ -46,10 +46,55 @@ FormplayerFrontend.module("SessionNavigate.MenuList", function (MenuList, Formpl
         },
     });
 
+    var getGridAttributes = function (tile) {
+        if (!tile) {
+            return null;
+        }
+        var rowStart = tile.gridY + 1;
+        var colStart = tile.gridX + 1;
+        var rowEnd = rowStart + tile.gridHeight;
+        var colEnd = colStart + tile.gridWidth;
+
+        return "grid-area: " + rowStart + " / " + colStart + " / " +
+            rowEnd + " / " + colEnd + ";";
+    };
+
+    function addStyleString(str) {
+        var node = document.createElement('style');
+        node.innerHTML = str;
+        document.body.appendChild(node);
+    }
+
     MenuList.CaseView = Marionette.ItemView.extend({
         tagName: "tr",
         template: "#case-view-item-template",
 
+        getTemplate: function () {
+            if (this.options.tiles) {
+                return "#case-tile-view-item-template";
+            } else {
+                return "#case-view-item-template";
+            }
+        },
+
+        initialize: function (options) {
+            this.tiles = options.tiles;
+            this.styles = options.styles;
+            for(var i = 0; i < this.tiles.length; i++) {
+                var tile = this.tiles[i];
+                if(tile === null){
+                    continue;
+                }
+                var fontSize = this.tiles[i].fontSize;
+                var fontString = "font-size: " + fontSize + ";"
+                var styleString = getGridAttributes(tile);
+                var tileId = "grid-style-" + i;
+                var formattedString = "." + tileId + " { " + styleString + " " + fontString + " } ";
+                addStyleString(formattedString);
```

Comment:
Dynamically adding page-level styles is interesting and rather unusual. It'd be more typical to use js to set the CSS properties on the individual tiles, so instead of this line, you'd have something like `$(".grid-style-" + i).css("font-size", fontSize).css("grid-style", getGridAttributes(tile))`.

Nitpick: given that the grid-style-X identifiers are unique, they're more like ids than classes.